### PR TITLE
Add setting for environment variables to send to Swift

### PR DESCRIPTION
**Pull request by adam-fowler**
Something like this @stevapple?

### CODE REVIEW

File: src/sourcekit-lsp/LanguageClientManager.ts [hunk 1]
```diff
@@ -247,7 +247,7 @@ export class LanguageClientManager {
         const sourcekit: langclient.Executable = {
             command: serverPath,
             args: lspConfig.serverArguments.concat(sdkArguments),
-            options: { env: { ...process.env, ...swiftRuntimeEnv() } },
+            options: { env: { ...configuration.swiftEnvironmentVariables, ...swiftRuntimeEnv() } },
```
Comment (stevapple) [inline src/sourcekit-lsp/LanguageClientManager.ts:250]: `process.env` should also be passed here

File: src/configuration.ts [hunk 2]
```diff
@@ -77,6 +77,12 @@ const configuration = {
     get buildArguments(): string[] {
         return vscode.workspace.getConfiguration("swift").get<string[]>("buildArguments", []);
     },
+    /** Environment variables to set when building */
+    get swiftEnvironmentVariables(): { [key: string]: string } {
```
Comment (stevapple) [inline src/configuration.ts:81]: I guess we can extract `Object.keys(swiftEnvironmentVariables).length > 0` in to a new computed property?

**Review by stevapple — CHANGES_REQUESTED** [review]
LGTM otherwise

**Comment by stevapple** [comment]
I think this could close #266 and I'll continue the remaining work based on #294

File: src/sourcekit-lsp/LanguageClientManager.ts (hunk 1 above)
Comment (adam-fowler) [reply src/sourcekit-lsp/LanguageClientManager.ts:250]: Just found we need to do the same with `execSwift`. It looks like we completely clear all other environment variables by setting one with `env` flags.

Comment (adam-fowler) [reply src/sourcekit-lsp/LanguageClientManager.ts]: Fixed

File: src/configuration.ts (hunk 2 above)
Comment (adam-fowler) [reply src/configuration.ts:81]: Not sure this is totally necessary, I do the same in execFile to make it more obvious why we are setting runtimePath env.

File: src/utilities/utilities.ts [hunk 3]
```diff
@@ -79,8 +79,15 @@ export async function execFile(
...
     );
-    if (configuration.runtimePath.length > 0 && customSwiftRuntime) {
-        options.env = { ...options.env, ...swiftRuntimeEnv(options.env) };
+    if (customSwiftRuntime) {
+        const runtimeEnv = swiftRuntimeEnv(options.env);
+        if (runtimeEnv && Object.keys(runtimeEnv).length > 0) {
+            options.env = { ...options.env, ...runtimeEnv };
+        }
+    }
+    // it options contain environment keys add them to process environment keys
```
Comment (0xTim) [inline src/utilities/utilities.ts:88]: Not spelling
```suggestion
    // if options contain environment keys add them to process environment keys
```

**Review by 0xTim — APPROVED** [review]
One minor spelling, otherwise LGTM

File: src/utilities/utilities.ts (hunk 3 above)
Comment (adam-fowler) [reply src/utilities/utilities.ts]: This code got deleted so no need to fix comment. It is now setup the same as in `execFileStreamOutput`

**Pull request merged**